java-analyzer-bundle.core/src/main/java/io/konveyor/tackle/core/internal/symbol/CustomASTVisitor.java

```diff
@@ -62,6 +68,64 @@ private boolean shouldVisit(ASTNode node) {
             (node.getStartPosition() + node.getLength());
     }
 
+    @Override
+    public boolean visit(MarkerAnnotation node) {
+        return visit((Annotation) node);
+    }
+
+    @Override
+    public boolean visit(NormalAnnotation node) {
+        return visit((Annotation) node);
+    }
+
+    @Override
+    public boolean visit(SingleMemberAnnotation node) {
+        return visit((Annotation) node);
+    }
+
+    private boolean visit(Annotation node) {
+        // There is a problem with trying to run shouldVisit() here because
+        // matches on annotations aren't directly on the annotation node,
+        // but on the annotated one (class, method, field, etc). So we can't
+        // use shouldVisit() to filter out nodes we don't want to visit.
+        // TODO: think of a better way to handle this
+        if (this.location != QueryLocation.ANNOTATION) {
+            return true;
+        }
+        try {
+            IAnnotationBinding binding = node.resolveAnnotationBinding();
+            if (binding != null) {
+                // get fqn
+                ITypeBinding declaringClass = binding.getAnnotationType();
+                if (declaringClass != null) {
+                    // Handle Erasure results
+                    if (declaringClass.getErasure() != null) {
+                        declaringClass = declaringClass.getErasure();
+                    }
+                    String fullyQualifiedName = declaringClass.getQualifiedName();
+                    // match fqn with query pattern
+                    if (fullyQualifiedName.matches(this.query)) {
+                        this.symbolMatches = true;
```

Comment:
What is happening here is that - visit() is called on every node in the comp unit. So if that class has an annotation matching the query elsewhere, symbolMatches will be set to true for all matches (annotations) in that class. Unlike methods, the match for annotation has the exact node, so it would be easier to compare whether the node for the given match is the one we are looking for here in visit() before setting symbolMatches to true. I think that is what's failing daytrader test cases -  there are files that have a real javax.ejb.EJB  annotation, but we are getting all annotations in that class as matches because there's one node that matches the query.